src/components/ChatEntry.js

```diff
@@ -1,22 +1,30 @@
-import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
 
 const ChatEntry = (props) => {
+  const buttonContent = props.liked ? '❤️' : '🤍';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={props.sender === 'Estragon' ? 'chat-entry remote' : 'chat-entry local'}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp} /></p>
+        <button className="like" onClick={() => props.onSetLike(props.id)}>{buttonContent}</button>
```

Comment:
👍 Nice inline callback to pass the id of the clicked message to the passed in handler.